lib/prometheus/PromClient.js

```diff
@@ -15,6 +15,73 @@ class PromClient {
     this.properties = properties;
     this.register = this.properties.register || REGISTER;
     this.metrics = {};
+    this.isChanged = false;
+
+    debug(this.properties);
+  }
+
+  _initiateCleaning(cleaning) {
+
+    if((!cleaning.rank && !cleaning.value) || !cleaning.interval) {
+      console.error("Misconfigured auto cleaning configuration, not cleaning up automatically");
+      return;
+    }
+
+    debug(`Auto cleaning for every ${cleaning.interval} seconds`);
+
+    setInterval(() => {
+
+      if (!this.isChanged) {
+        debug(`Not cleaning, there's no change in registry`);
+        return;
+      }
+
+      debug("Cleaning initiated...");
+
+      for(const key in this.metrics) {
+
+        const iterable = this.metrics[key]["hashMap"];
+
+        // Check filter method
+        if (cleaning.rank) {
+          this._cleanBasedOnRank(iterable);
+        }
+
+        if (cleaning.value) {
+          this._cleanBasedOnValue(iterable);
+        }
+      }
+
+      this.isChanged = false;
+    }, cleaning.interval * 1000);
+  }
+
+  _cleanBasedOnValue(metric) {
+
+    for (const key in metric) {
+      if (metric[key].value <= this.properties.cleaning.value) {
+        delete metric[key];
+      }
+    }
+  }
+
+  _cleanBasedOnRank(metric) {
+
+    const arr = [];
+    for (const key in metric) {
+      arr.push({key: key, value: metric[key]});
+    }
+
+    // Sort slice, map, and delete if not the same like the sorted key
+    const sortedKeys = arr.sort((a, b) => b.value.value - a.value.value)
+      .slice(0, this.properties.cleaning.rank)
+      .map(x => x.key);
```

Comment:
how performant is this, when running on millions of metrics?

Reply:
Checked on a million metrics, and it works okay. Complexity doesn't look too bad from the code.